src/plugin/ogc/ui/ogclayernodeui.js

```diff
@@ -45,15 +45,15 @@ plugin.ogc.ui.OGCLayerNodeUICtrl = function($scope, $element) {
 
   /**
    * The descriptor for this layer
-   * @type {plugin.ogc.OGCLayerDescriptor}
+   * @type {os.data.IDataDescriptor}
    * @private
    */
-  this.descriptor_ = /** @type {plugin.ogc.OGCLayerDescriptor} */ (
-      os.dataManager.getDescriptor(this.getLayerId()));
+  this.descriptor_ = os.dataManager.getDescriptor(this.getLayerId());
 
   var chooseTime = false;
 
-  if (this.descriptor_) {
+  if (os.implements(this.descriptor_, os.ui.ogc.IFeatureTypeDescriptor.ID)
+      && os.implements(this.descriptor_, os.ui.ogc.wms.IWMSLayer.ID)) {
     var featureType = this.descriptor_.getFeatureType();
     if (featureType) {
       chooseTime = this.descriptor_.hasTimeExtent() && featureType.getTimeColumns().length >= 2 &&
```

Comment:
I suspect the test for `hasTimeExtent` is really just making sure the start or end date column name isn't null ([this line](https://github.com/ngageoint/opensphere/blob/master/src/plugin/ogc/ogclayerdescriptor.js#L817) in the OGC descriptor).

I think we can replace that with the conditions in the linked line, and remove the `os.implements(this.descriptor_, os.ui.ogc.wms.IWMSLayer.ID)` test above.